bundles/model/org.eclipse.smarthome.model.thing/src/org/eclipse/smarthome/model/thing/internal/GenericThingProvider.xtend

```diff
@@ -263,7 +264,10 @@ class GenericThingProvider extends AbstractProvider<Thing> implements ThingProvi
 		]
 		channelDefinitions.forEach [
 			if (addedChannelIds.add(id)) {
-				channels += ChannelBuilder.create(new ChannelUID(thingUID, id), it.type.itemType).build
+			    val channelType = TypeResolver.resolve(it.channelTypeUID)
+			    if(channelType != null) {
+				    channels += ChannelBuilder.create(new ChannelUID(thingUID, id), channelType.itemType).build
+				}
```

Comment:
Shouldn't there be at least something logged, if a channel type could not be resolved?